### PR TITLE
Monitoring VMI Migration Phase Transition Times 

### PR DESCRIPTION
**Pull request by marceloamaral**
**What this PR does / why we need it**:

This PR introduces a new VMI Migration status field that tracks phase transition timestamps. It also adds two new histogram metrics to help track how long it takes for phase transitions to occur.

This is similar the VMI phase transition metric introduced in the PR #5766, but for the VMI Migration object.

The new histogram metric `kubevirt_vmi_migration_phase_transition_time_from_creation_seconds` tracks now long since creation it took to transition to the current phase a VMI Migration is in. This would be useful if you wanted to track how long it takes for VMI Migration to go from Creation to Succeeded (completed).

Using variations of that query, we can create graphs that indicate how long it takes for VMIs to reach running, and which phase the VMI spends the most time in.
image

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Addition of kubevirt_vmi_migration_phase_transition_time_from_creation_seconds metric to monitor how long it takes to transition a VMI Migration object to a specific phase from creation time.
```

### CODE REVIEW

File: pkg/monitoring/migration/vmi-migration-phase-transitions.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+)
+
+const (
+	transTimeErrFmt = "Error encountered during vmi migration transition time histogram calculation: %v"
+	transTimeFail   = "Failed to get a histogram for a vmi migration lifecycle transition times"
+)
+
+func getTransitionTimeSeconds(fromCreation bool, oldVMIMigration *v1.VirtualMachineInstanceMigration, newVMIMigration *v1.VirtualMachineInstanceMigration) (float64, error) {
```
[the remark this answers is not in the document]
Comment (marceloamaral) [reply pkg/monitoring/migration/vmi-migration-phase-transitions.go]: Oww, nice catch. I forgot to remove these variables, will fix it!

As for where the code should be, I'm not 100% sure either, but I have a better feeling having the migration metrics concentrated in the same place... But we could change it later...

File: pkg/virt-controller/watch/application.go [hunk 2]
```diff
@@ -345,6 +345,7 @@ func Execute() {
 	app.vmInformer = app.informerFactory.VirtualMachine()
 
 	app.migrationInformer = app.informerFactory.VirtualMachineInstanceMigration()
+	golog.Printf("\n ######## DEBUG app.migrationInformer :%v\n", app.migrationInformer)
```
Comment (rthallisey) [inline pkg/virt-controller/watch/application.go:348]: extra comment?

**Comment by rthallisey** [comment]
Code looks fine to me

File: pkg/virt-controller/watch/application.go (hunk 2 above)
Comment (marceloamaral) [reply pkg/virt-controller/watch/application.go]: Thanks, it was a let over!

**Pull request merged**